File: cloud-arm/src/main/java/com/sequenceiq/cloudbreak/cloud/arm/status/ArmStackStatus.java
package com.sequenceiq.cloudbreak.cloud.arm.status;

import com.sequenceiq.cloudbreak.cloud.model.ResourceStatus;

public class ArmStackStatus {

    private ArmStackStatus() {
    }

    public static ResourceStatus mapResourceStatus(String status) {
        switch (status) {
            case "Accepted":
                return ResourceStatus.IN_PROGRESS;
            case "Ready":
                return ResourceStatus.UPDATED;
            case "Canceled":
                return ResourceStatus.FAILED;
            case "Failed":
                return ResourceStatus.FAILED;
            case "Deleted":
                return ResourceStatus.DELETED;
            case "Succeeded":
                return ResourceStatus.CREATED;
            default:
                return ResourceStatus.IN_PROGRESS;

        }
    }
}

File: cloud-arm/src/main/java/com/sequenceiq/cloudbreak/cloud/arm/status/ArmStackStatus.kt
package com.sequenceiq.cloudbreak.cloud.arm.status

import com.sequenceiq.cloudbreak.cloud.model.ResourceStatus

object ArmStackStatus {

    fun mapResourceStatus(status: String): ResourceStatus {
        when (status) {
            "Accepted" -> return ResourceStatus.IN_PROGRESS
            "Ready" -> return ResourceStatus.UPDATED
            "Canceled" -> return ResourceStatus.FAILED
            "Failed" -> return ResourceStatus.FAILED
            "Deleted" -> return ResourceStatus.DELETED
            "Succeeded" -> return ResourceStatus.CREATED
            else -> return ResourceStatus.IN_PROGRESS
        }
    }
}